### PR TITLE
Support pip installed plugins

### PR DESCRIPTION
**Pull request by sagesmith-wf**
There are a few things still up for discussion here, such as documentation, how to enable/disable ect. but this provides a proof of concept and starts the discussion

#### Description of change
Arelle can load and run pip-installed plugins

#### Steps to Test
Pip install corresponding ixbrl-viewer https://github.com/Workiva/ixbrl-viewer/pull/381 branch into this branch

1. Pull both this and https://github.com/Workiva/ixbrl-viewer/pull/381
2. pip install <PATH_TO_ixbrl-viewer-repo>
3. python arelleCmdLine.py -f filing.htm --save-viewer ixbrl-report-viewer.html --viewer-url https://cdn-prod.wdesk.com/ixbrl-viewer/1.1.44/ixbrlviewer.js

**review**:
@Arelle/arelle

### CODE REVIEW

**Comment by aviary3-wk** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://wiki.atl.workiva.net/pages/viewpage.action?pageId=42762752#SecurityInsightFramework&Tooling-ConfiguringAviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

File: arelle/PluginManager.py [hunk 1]
```diff
@@ -71,6 +72,11 @@ def init(cntlr: Cntlr, loadPluginConfig: bool = True) -> None:
         pluginConfigChanged = False # don't save until something is added to pluginConfig
     modulePluginInfos = {}  # dict of loaded module pluginInfo objects by module names
     pluginMethodsForClasses = {} # dict by class of list of ordered callable function objects
+    unloaded_arelle_plugins = entry_points(group='arelle.plugin')
+    if unloaded_arelle_plugins:
+        for unloaded_plugin in unloaded_arelle_plugins:
+            plugin_url = unloaded_plugin.load()
+            addPluginModule(plugin_url())
```
Comment (austinmatherne-wk) [inline arelle/PluginManager.py:79]: While the plugin architecture should permit loading pip installed plugins, I think it should still require enabling the plugin via the command line or GUI

Comment (sagesmith-wf) [reply arelle/PluginManager.py:79]: Discussed a bit offline. I think we landed on passing the plugin name to enable it still

File: arelle/PluginManager.py [hunk 2]
```diff
@@ -493,7 +494,15 @@ def pluginClassMethods(className: str) -> Iterator[Callable[..., Any]]:
             yield method
 
 def addPluginModule(url):
-    moduleInfo = moduleModuleInfo(url)
+    moduleInfo = None
+    unloaded_arelle_plugins = entry_points(group='arelle.plugin', name=url)
+    if unloaded_arelle_plugins:
+        if len(unloaded_arelle_plugins) != 1:
+            raise Exception(f'multiple pip installed plugins with name {url} in group arelle.plugin')
```
Comment (austinmatherne-wk) [inline arelle/PluginManager.py:501]: Looking at the code path there doesn't appear to be a try except block around this. Can you look at logging an error instead? We typically try to fail gracefully when plugins can't be loaded.

**Review by derekgengenbacher-wf — APPROVED** [review]

**Review by allenbross-wf — APPROVED** [review]

**Review by austinmatherne-wk — APPROVED** [review]

**Comment by austinmatherne-wk** [comment]
# Verified

* pip installed plugins can be loaded from the command line
* existing plugins continue to work with both the command line and GUI

**Pull request merged**